gui/src/main/scala/co/uproot/abandon/RegReport.scala

```diff
@@ -30,34 +36,49 @@ object RegUIReport extends UIReport {
     new TreeView(reportRoot) {
       styleClass += styleClassName
       onKeyTyped = { e: KeyEvent =>
-        if (e.character equals "\r") {
-          val selectedItemOpt = selectionModel().getSelectedItems().headOption
-          selectedItemOpt foreach { selectedItem =>
-            // println(e)
-            val txStage = new Stage() {
-              scene = new Scene(800, 600) {
-                root = new ScrollPane {
-                  content = TxnUIReport.mkTxnView(getNestedTxns(selectedItem))
-                }
-                stylesheets += "default_theme.css"
-              }
-              initModality(Modality.ApplicationModal)
-              title = "Transactions"
+        if (e.character.equals("\r") && !selectionModel().isEmpty) {
+          // println(e)
+          selectedItem = selectionModel().getSelectedItem
+          // show all transactions of the parent of the selected item if the latter has any transactions
+          // this will allow for an update of the transaction view when an input file changes
+          if (selectedItem.getValue.txns.nonEmpty) selectedItem = selectedItem.getParent
```

Comment:
I'm aware of the problem, and you're right - it's not pretty. But I didn't know how to solve the problem that arose when selecting specific transactions for display - the update didn't work then. It only works for the root and the date range nodes.
So, your solution proposal isn't too bad at all - I'll look into it.